hangman.py

```diff
@@ -152,62 +253,118 @@ def show_possible_matches(my_word):
              that has already been revealed.
 
     '''
-    # FILL IN YOUR CODE HERE AND DELETE "pass"
-    pass
 
+    list = []
+    for word in wordlist:
+        if match_with_gaps(my_word, word):
+            list.append(word)
+
+    if len(list) != 0:
+        print(' '.join(list))
+    else:
+        print('No matches found')
 
 
 def hangman_with_hints(secret_word):
```

Comment:
### DRY
Don't repeat yourself. For make your program more productive, do the following:

**Remove** function `hangman_with_hints `that are related to hints.
**Add** additional parameters to function hangman to transfer game mode.
**Create** a game mode check. Create a separate code block for the game mode with hints
**Use** functions **check_warnings** and **check_guesses** to reduce the code and make it correct

Try to avoid such mistakes.